src/main.rs

```diff
@@ -23,18 +53,54 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         .add_source(config::File::with_name("Config"))
         .set_override("homeserver_url", args.homeserver)?
         .set_override("output_dir", args.output_dir)?
+        .set_override("create", args.create)?
+        .set_override("delete", args.delete)?
+        .set_override("run", args.run)?
+        .set_override_option("timestamp", args.timestamp)?
+        .set_override("user_count", args.amount)?
+        .set_override_option("users_filename", Some(args.users_filename))?
         .build()?;
 
     let config = config.try_deserialize::<Configuration>();
     match config {
         Ok(config) => {
-            let mut state = State::new(config);
-            state.run().await;
+            match config {
+                Configuration { create: true, .. } => create_users(config).await,
+                Configuration { delete: true, .. } => {
+                    delete_users(config).await;
+                }
+                Configuration { run: true, .. } => {
+                    run_state(config).await;
+                }
+                _ => println!("One of create delete or run modes must be selected"),
+            };
         }
         Err(e) => {
             println!("Couldn't parse config {}", e);
         }
-    };
+    }
 
     Ok(())
 }
+
+async fn run_state(config: Configuration) {
+    let mut state = State::new(config);
+    state.run().await
+}
+
+async fn create_users(config: Configuration) {
+    let mut state = State::new(config);
+    state.create_users().await
+}
+
+async fn delete_users(config: Configuration) {
+    match config.timestamp {
+        Some(_) => {
+            let mut state = State::new(config);
+            state.run().await
```

Comment:
This should be the delete. Let's remove the run and instead write a `TODO` to avoid confusion.